File: problems/28-find-the-index-of-the-first-occurrence-in-a-string.ts
export function strStr(haystack: string, needle: string): number {
  if (!haystack.length && !needle.length) return 0;
  // return haystack.indexOf(needle);
  for (let i = 0; i <= haystack.length - needle.length; i++) {
    if (haystack.slice(i, i + needle.length) === needle) {
      return i;
    }
  }

  return -1;
}

File: go/problems/28_find_the_index_of_the_first_occurrence_in_a_string.go
package main

func strStr(haystack string, needle string) int {
	if len(needle) > len(haystack) {
		return -1
	}

	if len(needle) == len(haystack) {
		return 0
	}

	for i := 0; i <= len(haystack)-len(needle); i++ {
		if needle == haystack[i:i+len(needle)] {
			return i
		}
	}

	return -1
}
